Mathlib/Geometry/Manifold/VectorField.lean

```diff
@@ -0,0 +1,317 @@
+      hv mdifferentiableWithinAt_id
+  /- To prove that `ϕ` depends smoothly on `x`, we use that the derivative depends smoothly on `x`
+  (this is `ContMDiffWithinAt.mfderivWithin_const`), and that taking the inverse is a smooth
+  operation at an invertible map. -/
+  -- the derivative in coordinates depends smoothly on the point
+  have : MDifferentiableWithinAt I 𝓘(𝕜, E →L[𝕜] E')
+      (fun (x : M) ↦ ContinuousLinearMap.inCoordinates
+        E (TangentSpace I (M := M)) E' (TangentSpace I' (M := M'))
+        x₀ x (f x₀) (f x) (mfderivWithin I I' f s x)) s x₀ :=
+    ((hf.of_le hmn).mfderivWithin_const le_rfl hx₀ hs).mdifferentiableWithinAt le_rfl
+  -- therefore, its inverse in coordinates also depends smoothly on the point
+  have : MDifferentiableWithinAt I 𝓘(𝕜, E' →L[𝕜] E)
+      (ContinuousLinearMap.inverse ∘ (fun (x : M) ↦ ContinuousLinearMap.inCoordinates
+        E (TangentSpace I (M := M)) E' (TangentSpace I' (M := M'))
+        x₀ x (f x₀) (f x) (mfderivWithin I I' f s x))) s x₀ := by
+    apply MDifferentiableAt.comp_mdifferentiableWithinAt _ _ this
+    apply ContMDiffAt.mdifferentiableAt _ le_rfl
+    apply ContDiffAt.contMDiffAt
+    apply IsInvertible.contDiffAt_map_inverse
+    rw [inCoordinates_eq (FiberBundle.mem_baseSet_trivializationAt' x₀)
+      (FiberBundle.mem_baseSet_trivializationAt' (f x₀))]
+    exact isInvertible_equiv.comp (hf'.comp isInvertible_equiv)
+  -- the inverse in coordinates coincides with the in-coordinate version of the inverse,
+  -- therefore the previous point gives the conclusion
+  apply this.congr_of_eventuallyEq_of_mem _ hx₀
+  have A : (trivializationAt E (TangentSpace I) x₀).baseSet ∈ 𝓝[s] x₀ := by
+    apply nhdsWithin_le_nhds
+    apply (trivializationAt _ _ _).open_baseSet.mem_nhds
+    exact FiberBundle.mem_baseSet_trivializationAt' _
+  have B : f ⁻¹' (trivializationAt E' (TangentSpace I') (f x₀)).baseSet ∈ 𝓝[s] x₀ := by
+    apply hf.continuousWithinAt.preimage_mem_nhdsWithin
+    apply (trivializationAt _ _ _).open_baseSet.mem_nhds
+    exact FiberBundle.mem_baseSet_trivializationAt' _
+  filter_upwards [A, B] with x hx h'x
+  simp only [Function.comp_apply]
+  rw [inCoordinates_eq hx h'x, inCoordinates_eq h'x (by exact hx)]
+  simp only [inverse_equiv_comp, inverse_comp_equiv, ContinuousLinearEquiv.symm_symm, ϕ]
+  rfl
+
+lemma _root_.MDifferentiableWithinAt.mpullbackWithin_vectorField_inter_of_eq
+    (hV : MDifferentiableWithinAt I' I'.tangent
+      (fun (y : M') ↦ (V y : TangentBundle I' M')) t y₀)
+    (hf : ContMDiffWithinAt I I' n f s x₀) (hf' : (mfderivWithin I I' f s x₀).IsInvertible)
+    (hx₀ : x₀ ∈ s) (hs : UniqueMDiffOn I s) (hmn : 2 ≤ n) (h : y₀ = f x₀) :
+    MDifferentiableWithinAt I I.tangent
+      (fun (y : M) ↦ (mpullbackWithin I I' f V s y : TangentBundle I M)) (s ∩ f⁻¹' t) x₀ := by
+  subst h
+  exact hV.mpullbackWithin_vectorField_inter hf hf' hx₀ hs hmn
+
+/-- The pullback of a `C^m` vector field by a `C^n` function with `m + 1 ≤ n` is `C^m`.
```

Comment:
This doc-string doesn't match the statement. Can you update it, please?
(The statement of the doc-string would also be nice, though. Is that part of a future PR?)

Reply:
Yes, it's part of the next PR. Copy-paste mistake, presumably. Fixed.